lib/Bio/EnsEMBL/DataCheck/Checks/CheckTableSizes.pm

```diff
@@ -0,0 +1,89 @@
+=head1 LICENSE
+
+Copyright [2018-2019] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the 'License');
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+package Bio::EnsEMBL::DataCheck::Checks::CheckTableSizes;
+
+use warnings;
+use strict;
+
+use Moose;
+use Test::More;
+use Bio::EnsEMBL::DataCheck::Test::DataCheck;
+use Bio::EnsEMBL::DataCheck::Utils qw/ array_diff /;
+
+extends 'Bio::EnsEMBL::DataCheck::DbCheck';
+
+use constant {
+  NAME           => 'CheckTableSizes',
+  DESCRIPTION    => 'Tables must be populated and not differ significantly in row numbers',
+  GROUPS         => ['compara', 'compara_tables'],
+  DATACHECK_TYPE => 'advisory',
+  DB_TYPES       => ['compara'],
+  TABLES         => ['CAFE_gene_family', 'CAFE_species_gene', 'conservation_score', 'constrained_element', 'dnafrag', 'dnafrag_region', 'exon_boundaries', 'external_db', 'family', 'family_member', 'gene_align', 'gene_align_member', 'gene_member', 'gene_member_hom_stats', 'gene_member_qc', 'gene_tree_node', 'gene_tree_node_attr', 'gene_tree_node_tag', 'gene_tree_object_store', 'gene_tree_root', 'gene_tree_root_attr', 'gene_tree_root_tag', 'genome_db', 'genomic_align', 'genomic_align_block', 'genomic_align_tree', 'hmm_annot', 'hmm_curated_annot', 'hmm_profile', 'homology', 'homology_member', 'mapping_session', 'member_xref', 'method_link', 'method_link_species_set', 'method_link_species_set_attr', 'method_link_species_set_tag', 'ncbi_taxa_name', 'ncbi_taxa_node', 'other_member_sequence', 'peptide_align_feature', 'seq_member', 'seq_member_projection', 'seq_member_projection_stable_id', 'sequence', 'species_set', 'species_set_header', 'species_set_tag', 'species_tree_node', 'species_tree_node_attr', 'species_tree_node_tag', 'species_tree_root', 'stable_id_history', 'synteny_region']
+};
+
+sub tests {
+  my ($self) = @_;
+  
+  my $curr_dba = $self->dba;
+  my $curr_helper = $curr_dba->dbc->sql_helper;
+  my $prev_dba = $self->registry->get_DBAdaptor('compara_prev', 'compara');
```

Comment:
`compara_prev` won't work on the DC service because the latter doesn't use our different registry file. Instead the DC framework provides `$self->get_old_dba`. 
```suggestion
  my $prev_dba = $self->registry->get_DBAdaptor('compara_prev', 'compara') || $self->get_old_dba;
```

Reply:
Yes, this was why I was doing it the way I was before, I thought it made sense to force it to use the registry file we will already have produced. I also thought with providing our own registry, it gives the ability to be more flexible with which previous db is used. I will stick this in anyway.